### PR TITLE
Ensure handymen wait for patients to leave

### LINKED ISSUE #70: Machines can be repaired even if it is in use
If a patient is using a machine, a handyman can still repair this machine (like the scanner) and this gives a strange animation (of a not-working and a working machine in one.)

Of course the handyman should wait with repairing until the machine is not in use.

### PR DESCRIPTION
**Pull request by SilPho**
*Fixes #70 *

**Describe what the proposed change does**
- Handymen waiting to repair a machine will now meander in the room if a patient is in the same room. This fixes the bug where the machine is repaired while it is in use.

This is the same behaviour that the original game has.

### CODE REVIEW

**Comment by lewri** [comment]
Does this one still allow the handyman to enter the room? I know currently the handyman can wait in the room if things are going outside of the realms of the bug.

**Comment by SilPho** [comment]
Yes, this change means that they just wait inside the room for longer while the patient is being treated or diagnosed. Without this change the handyman will wait in the room if the machine is in use when he gets there, but if the patient is changing behind a screen or just hasn't quite made it to the machine yet. With the fix the handyman will wait until the patient, if any, is leaving.

Technically this would break in rooms with multiple patients, but none of those have repairable machines in them.

File: CorsixTH/Lua/entities/machine.lua [hunk 1]
```diff
@@ -237,7 +237,8 @@ function Machine:createHandymanActions(handyman)
   end
 
   local meander_loop_callback = --[[persistable:handyman_meander_repair_loop_callback]] function()
-    if not self.user then
+    -- Wait until the machine is not in use and not about to be used
+    if not self.user and (not self:getRoom():getPatient() or self:getRoom():getPatient().isLeaving) then
```
Comment (lewri) [inline CorsixTH/Lua/entities/machine.lua:241]: You call the two functions twice here. You'd probably be better making a `local patient = self:getRoom():getPatient()` then in the if:
` if not self.user and (not patient or patient.isLeaving) then`

Comment (SilPho) [reply CorsixTH/Lua/entities/machine.lua]: Thanks. Not sure why I didn't do that sooner

**Review by lewri — APPROVED** [review]
I'm ok with this

File: CorsixTH/Lua/entities/machine.lua [hunk 2]
```diff
@@ -237,7 +237,9 @@ function Machine:createHandymanActions(handyman)
   end
 
   local meander_loop_callback = --[[persistable:handyman_meander_repair_loop_callback]] function()
-    if not self.user then
+    -- Wait until the machine is not in use and not about to be used
+    local patient = self:getRoom():getPatient()
+    if not self.user and (not patient or patient.isLeaving) then
```
Comment (lewri) [inline CorsixTH/Lua/entities/machine.lua:242]: My oversight, I think this one needs to be `patient:isLeaving()`

Calling `patient.isLeaving` just confirms that a function exists for the patient.

**Pull request merged**